### PR TITLE
Generate scope id from file name

### PR DESCRIPTION
**Pull request by dsnjunior**
## Changes
- Generate Scope IDs deterministically. This means IDs will match between development and production builds.

### CODE REVIEW

**Comment by dsnjunior** [comment]
Don't know how to test this one 😅

File: packages/scope/index.js [hunk 1]
```diff
@@ -32,6 +32,14 @@ export function scope(id) {
 	};
 }
 
+/**
+ * @param {string} path
+ * @returns {string}
+ */
+function createScopeHash(path) {
+	return createHash("sha1").update(path).digest("hex");
```
Comment (dsnjunior) [inline packages/scope/index.js:40]: Maybe?
```ts
return `${path.split('/').pop()}-${createHash("sha1").update(path).digest("hex")}`;
```

Comment (bholmesdev) [reply packages/scope/index.js:40]: Oh, I'm curious why the `split()` would be better? Using a `/` seems error-prone with windows machines, and I'm not sure why we want the last relative path segment 🤷

Comment (bholmesdev) [reply packages/scope/index.js:40]: Sending this to a couple Astro core members to make sure on security. I _think_ this code looks good, just want to be sure!

Comment (dsnjunior) [reply packages/scope/index.js:40]: Fair points! I agree we don't need this split thing. And, now that you said, I'm not sure about the safety 😅

Comment (dsnjunior) [reply packages/scope/index.js]: Done it!

**Comment by bholmesdev** [comment]
Ah, I see you attempted a force puch to resolve conflicts. I may have botched the commit history in a recent merge, so I'll take care of that 😅

File: .changeset/orange-countries-think.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"vite-plugin-simple-scope": minor
```
Comment (bholmesdev) [inline .changeset/orange-countries-think.md:2]:
```suggestion
"vite-plugin-simple-scope": major
```

**Pull request merged**